engine/access/rest/http/request/get_transaction.go

```diff
@@ -64,6 +64,57 @@ func (g *GetTransaction) Build(r *common.Request) error {
 	return err
 }
 
+// GetTransactionsByBlockID represents a request to get transactions by
+// a block ID, and contains the parsed and validated input parameters.
+type GetTransactionsByBlockID struct {
+	TransactionOptionals
+	BlockHeight   uint64
+	ExpandsResult bool
+}
+
+// NewGetTransactionsByBlockIDRequest extracts necessary variables from the provided request,
+// builds a GetTransactionsByBlockID instance, and validates it.
+//
+// All errors indicate a malformed request.
+func NewGetTransactionsByBlockIDRequest(r *common.Request) (*GetTransactionsByBlockID, error) {
+	req, err := parseGetTransactionsByBlockID(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return req, nil
+}
+
+// parseGetTransactionsByBlockID parses raw query and body parameters from an incoming request
+// and constructs a validated GetTransactionsByBlockID instance.
+//
+// All errors indicate the request is invalid.
+func parseGetTransactionsByBlockID(r *common.Request) (*GetTransactionsByBlockID, error) {
+	var req GetTransactionsByBlockID
+	err := req.TransactionOptionals.Parse(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var height Height
+	err = height.Parse(r.GetQueryParam(blockHeightQuery))
+	if err != nil {
+		return nil, err
+	}
+	req.BlockHeight = height.Flow()
+	req.ExpandsResult = r.Expands(resultExpandable)
+
+	if req.BlockHeight == EmptyHeight && req.BlockID == flow.ZeroID {
+		req.BlockHeight = FinalHeight
+	}
+
+	if req.BlockID != flow.ZeroID && req.BlockHeight != EmptyHeight {
```

Comment:
nit: make the 2 checks symmetrical so it's easier to read
```suggestion
	if req.BlockHeight != EmptyHeight && req.BlockID != flow.ZeroID {
```